packages/backend-modules/access/lib/grants.js

```diff
@@ -589,6 +625,33 @@ const beginGrant = async (grant, payload, recipient, pgdb) => {
   return { ...result, granter, recipient, campaign }
 }
 
+const findEmptyRecommendations = async (campaign, pgdb) => {
+  const beginBefore = moment()
+
+  Object.keys(campaign.emailRecommendations).forEach((unit) =>
+    beginBefore.subtract(campaign.emailRecommendations[unit], unit),
+  )
+
+  debug('findEmptyRecommendations', { campaign: campaign.id })
+
+  const grants = await pgdb.public.accessGrants.find({
+    accessCampaignId: campaign.id,
+    'beginAt <': beginBefore,
+    recommendationsAt: null,
+  })
```

Comment:
```suggestion
  const grants = await pgdb.public.accessGrants.find({
    accessCampaignId: campaign.id,
    'beginAt <': beginBefore,
    invalidatedAt: null,
    recommendationsAt: null,
  })
```

Would add `invalidatedAt: null` here to return valid grants, thus sending recommendations only to users which do have access.